guides/decoding-encoding/getBlockById.js

```diff
@@ -0,0 +1,37 @@
+const { getClient } = require('./api-client');
+const assert = require("assert");
+
+if (process.argv.length < 3) {
+	console.error("Please provide the block ID to be decoded.")
+	process.exit(1);
+}
+const blockId = process.argv[2];
+
+getClient().then((client) => {
+	// Returns the block
+	client.invoke("chain_getBlockByID", {
+		id: blockId
+	}).then(block => {
```

Comment:
here just to clarify, block is returned in the JSON format

Reply:
yea `toJSON` will also decode the internal params with metadata

Reply:
This is what i meant with confusing formats: there are too many and it is unclear what is used for what. For example, if someone says "The Tx is returned in JSON format" this is not really precise, and it needs a follow up question "Which JSON format?" Also currently, I am not really aware what is the exact use case for every existing format

Reply:
so there are global concept of block header / transaction / asset / event where `params` or `data` is hex string. This is basically because each app can decide how to use those parameters.
In the endpoint, for block/transaction always global concept are returned (in JSON format).

However, in client, it is internally to decode further using metadata for user. 
For example, if you use `client.block.get(id)` it will actually return the fully decoded version, but here you are using raw endpoint